pallets/liquid-staking/src/migrations.rs

```diff
@@ -13,20 +13,81 @@
 // limitations under the License.
 
 pub mod v3 {
-    use crate::Config;
+    use crate::{
+        types::{MatchingLedger, ReservableAmount},
+        BalanceOf, Config, MatchingPool, StakingLedgerCap, StorageVersion,
+    };
     use frame_support::pallet_prelude::*;
+    use frame_support::{
+        generate_storage_alias, log,
+        traits::{tokens::Balance as BalanceT, Get},
+        weights::Weight,
+    };
+
+    generate_storage_alias!(LiquidStaking, MarketCap => Value<u128,ValueQuery>);
+
+    #[derive(Copy, Clone, Eq, PartialEq, Default, Encode, Decode, RuntimeDebug, TypeInfo)]
+    pub struct OldMatchingLedger<Balance: BalanceT> {
+        /// The total stake amount in one era
+        pub total_stake_amount: Balance,
+        /// The total unstake amount in one era
+        pub total_unstake_amount: Balance,
+    }
 
     #[cfg(feature = "try-runtime")]
     pub fn pre_migrate<T: Config>() -> Result<(), &'static str> {
+        log::info!("MarketCap.get()? {:?}", MarketCap::get());
+        log::info!("MatchingPool.get()? {:?}", MatchingPool::<T>::get());
+        assert!(MarketCap::exists(), "MarketCap storage item not found!");
         Ok(())
     }
 
     pub fn migrate<T: Config>() -> Weight {
-        0
+        if StorageVersion::<T>::get() == crate::Versions::V2 {
+            log::info!("migrating liquidStaking to Versions::V3",);
+            // 1. from MarketCap to StakingLedgerCap
+            let cap = MarketCap::get();
+            MarketCap::kill();
+            StakingLedgerCap::<T>::put(cap);
```

Comment:
since we need to do migration storage version v3 for both heiko & parallel and they have different staking ledger cap, I think we can left it to be empty and set it once runtime upgrade succeeded